tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalDataInfo.java

```diff
@@ -2639,6 +2639,47 @@ public int getCoverageValueOld(String xpath, ULocale loc) {
         return Level.OPTIONAL.getLevel(); // If no match then return highest possible value
     }
 
+    // The following is for mapping language to the explicit script and region codes in
+    // the locale IDs for CLDR locales. We don't need to worry about the unmarked default
+    // script or region since they are already supplied by the <languageData>.
+    private Map<String, BasicLanguageData> doMapLanguagesToScriptsRegion() {
+        Map<String, BasicLanguageData> langToScriptsRegions = new TreeMap<>();
+        org.unicode.cldr.util.Factory factory = CLDRConfig.getInstance().getCldrFactory();
+        for (CLDRLocale locale : factory.getAvailableCLDRLocales()) {
+            String language = locale.getLanguage();
+            if (language.length() == 0 || language.equals("root")) {
+                continue;
+            }
+            BasicLanguageData scriptsAndRegions = langToScriptsRegions.get(language);
+            if (scriptsAndRegions == null) {
+                scriptsAndRegions = new BasicLanguageData();
+                langToScriptsRegions.put(language, scriptsAndRegions);
+            }
+            String script = locale.getScript();
+            if (script.length() > 0) {
+                scriptsAndRegions.addScript(script);
+            }
+            String region = locale.getCountry();
+            if (region.length() > 0 && region.length() < 3) { // per CLDR TC, do not want 001, 419 etc.
```

Comment:
this works, though perhaps a character text (alpha vs digits) would be more robust?

Reply:
> It makes some sense, but shouldn't this also be a test to make sure we update supplemental data? Maybe that's already the case and it's an exhaustive test?

I believe that updating supplemental languageData would not actually help here; the languageData has somewhat different intended goals and usages. Many of the script and regions that would be added to coverage per the code here are already in languageData as secondary scripts or regions, but such secondary items are excluded for getTargetScripts and getTargetTerritories. But we don't want to just change getTargetScripts and getTargetTerritories to include the secondary items because that would also add many script and regions for which we do *not* have locales.

Reply:
> this works, though perhaps a character text (alpha vs digits) would be more robust?

Not sure it makes it more robust unless I added both a length and character type check. But not sure it is necessary and is (just slightly) less performant.